src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseKey.cs

```diff
@@ -207,7 +207,9 @@ internal int Sign(ReadOnlySpan<byte> toBeSigned, Span<byte> destination)
 #pragma warning disable SYSLIB5006
                 case KeyType.MLDsa:
                     Debug.Assert(_mldsaKey != null);
-                    return _mldsaKey.SignData(toBeSigned, destination);
+                    Span<byte> mldsaSignature = destination.Slice(0, _mldsaKey.Algorithm.SignatureSizeInBytes);
+                    _mldsaKey.SignData(toBeSigned, mldsaSignature);
```

Comment:
Is `destination` guaranteed big enough here?  If not, is this changing what exception is thrown?